flink-core/src/main/java/org/apache/flink/configuration/TaskManagerOptions.java

```diff
@@ -79,6 +79,16 @@
 			.defaultValue(false)
 			.withDescription("Whether to kill the TaskManager when the task thread throws an OutOfMemoryError.");
 
+	/**
+	 * Whether the task manager should attempt to gracefully shutdown itself on errors.
+	 */
+	@Documentation.Section(Documentation.Sections.ALL_TASK_MANAGER)
+	public static final ConfigOption<Boolean> GRACEFUL_SHUTDOWN_ON_ERROR =
+		key("taskmanager.graceful-shutdown-on-error")
```

Comment:
I think it would be nice to make this behaviour configurable for all Flink processes.

Reply:
The scope is currently all task manager processes but it makes sense to extend it to all entry points.